src/vector.c

```diff
@@ -34,6 +34,14 @@
 #define STATE_DIMS(x) (ARR_DIMS(x)[0] - 1)
 #define CreateStateDatums(dim) palloc(sizeof(Datum) * (dim + 1))
 
+#if defined _MSC_VER /* MSVC only */
+#define __attribute__(x)
+#elif defined __APPLE__ /* Apple/OSX only */
+#define target_clones(...)
+#elif !defined __has_attribute || !__has_attribute(target_clones) || !defined __gnu_linux__ /* target_clones not supported */
+#define target_clones(...)
+#endif
```

Comment:
These, now, handle the following situations:

- for MS Visual C compiler, `__attribute__` is defined to be no-op,
- for Apple targets, `target_clones` is defined to be no-op, and,
- whenever `target_clones` is not found as an attribute or `__gnu_linux__` is not defined (i.e., for Alpine), `target_clones` is defined to be no-op.

You can verify the implementation for `l2_distance_impl` on Godbolt: https://godbolt.org/z/jzMjKbhKe

I have also tested the compilation in `alpine:latest` images. GCC builds _without_ the SIMD support whereas Clang 16 builds _with_ it. However, when I try to run the compiled code, I receive an error as mentioned in llvm/llvm-project#64631, which is closed by llvm/llvm-project-release-prs#615. In fact, when I test the code in `alpine:edge`, instead, GCC still builds the code _without_ the SIMD support whereas Clang 17 builds _with_ it, and I am able to see the benefits (speed improvements of up to 70%, again in pure C benchmarks).